tests/cnf/core/network/metallb/tests/bgp-gracefulrestart.go

```diff
@@ -205,6 +373,59 @@ func gracefulRestartTest(
 	Expect(err).ToNot(HaveOccurred(), "Unexpectedly succeeded in curling the Nginx pod")
 }
 
+func gracefulRestartTestWithBFD(
+	extTrafficPolicy corev1.ServiceExternalTrafficPolicyType,
+	destIPaddress,
+	ipv4metalLbIP string,
+	frrPod *pod.Builder,
+	ipAddressPool *metallb.IPAddressPoolBuilder,
+	workerLabelMap map[string]string) {
+	By("Creating a MetalLB service")
+	setupMetalLbService(netparam.IPV4Family, ipAddressPool, extTrafficPolicy)
+
+	By("Checking that routes have been added to the BGP table")
+	verifyRoutesAmount(frrPod, 1)
+
+	By("Running http check before the fail over test")
+	validateCurlToNginxPod(frrPod, ipv4metalLbIP, destIPaddress)
+
+	By("Deleting the MetalLB CR to remove all frr-k8s pods and simulate a GR scenario (e.g., during an upgrade)")
```

Comment:
Could you remind me why you prefer to delete the MetalLB CR rather than deleting the frr-k8s pod which is better "emulation" of an upgrade?

Reply:
1. Delleting frr-k8s pods might be too quick to properly verify all checks (BGP down, BGP routes, traffic validation).
2. There is a validation in place that checks if the routes are removed after the GR timeout timer.

Reply:
Did you try out the "might be too quick"?

The reason I am asking is that we should eventually add proper cleanup when the MetalLB CR is deleted, if that happen this test will regress

Reply:
10 sec. This is not enough to run all checks